pandas/core/arrays/base.py

```diff
@@ -138,6 +140,34 @@ def nbytes(self):
     # ------------------------------------------------------------------------
     # Additional Methods
     # ------------------------------------------------------------------------
+    def astype(self, dtype, copy=True):
+        """Cast to a NumPy array with 'dtype'.
+
+        The default implementation only allows casting to 'object' dtype.
+
+        Parameters
+        ----------
+        dtype : str or dtype
+            Typecode or data-type to which the array is cast.
+        copy : bool, default True
+            Whether to copy the data, even if not necessary. If False,
+            a copy is made only if the old dtype does not match the
+            new dtype.
+
+        Returns
+        -------
+        array : ndarray
+            NumPy ndarray with 'dtype' for its dtype.
+        """
+        np_dtype = np.dtype(dtype)
+
+        if np_dtype != 'object':
```

Comment:
use is_object_dtype

Reply:
why do you even need this?

Reply:
You mean the check for object? Fair point. If the underlying object supports conversion to whatever format, we should allow it.

```python
In [1]: import pandas_ip as ip

In [2]: ip.IPAddress([1, 2, 3]).astype(object)
Out[2]:
array([IPv4Address('0.0.0.1'), IPv4Address('0.0.0.2'),
       IPv4Address('0.0.0.3')], dtype=object)

In [3]: ip.IPAddress([1, 2, 3]).astype(int)
Out[3]: array([1, 2, 3])

```

Which simplifies things nicely!